feat(agent): add support for Windows Agent

iurly opened the pull request:
## What this PR does / why we need it:
This is PR adds support for the Windows Agent

## Checklist

<!-- [Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->

- [x] Title of the PR starts with type and scope, (e.g. `feat(agent,node-analyzer,sysdig-deploy):`)
- [x] Chart Version bumped for the respective charts
- [x] Variables are documented in the README.md (or README.tpl in some charts)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

<!-- Check Contribution guidelines in README.md for more insight. -->

iurly committed:
wip: windows chart

iurly committed:
wip2

iurly committed:
remove duplicates

iurly committed:
implement nodeSelector

iurly committed:
cleanup

iurly committed:
default imagepullPolicy

iurly committed:
remove separate configmap

iurly committed:
add secrets, though unused

iurly committed:
Merge remote-tracking branch 'origin/main' into SMAGENT-6917-add-support-for-windows-agent

AlbertoBarba commented on charts/agent/templates/daemonset-windows.yaml:
```diff
@@ -0,0 +1,69 @@
+          {{- else }}
+          secretName: {{ include "agent.accessKeySecret" . }}
+          {{- end }}
+      nodeSelector:
```
@iurly would it make sense to add
```yaml
"kubernetes.io/os": linux
```
in:
- charts/agent/templates/daemonset.yaml
- charts/agent/templates/deployment.yaml
?

iurly committed:
guard rendering of customerid for temporary testing

iurly committed:
Merge remote-tracking branch 'origin/main' into SMAGENT-6917-add-support-for-windows-agent

iurly committed:
temporarily bump chart version

iurly committed:
bump agent version in sysdig-deploy chart

iurly committed:
rework values for windows

- introduce windows top-level object with all settings
- fix agent-windows.* helpers
- add serviceAccountName (same as linux)
- fix handling of windows.image.registry
- move nodeSelector from windowsDaemonset. to windows.
- add windows.affinity and windows.tolerations
- add windows.resources
- use windows.image.tag as app.kubernetes.io/version

iurly committed:
fix imagePullSecrets

iurly committed:
remove accesskey from configmap

iurly committed:
bump sysdig-deploy

iurly committed:
implement windows.enabled

iurly committed:
update README.md with windows entries

AlbertoBarba commented on charts/agent/templates/daemonset-windows.yaml:
```diff
@@ -0,0 +1,74 @@
+          secretName: {{ template "agent.fullname" . }}
+          {{- else }}
+          secretName: {{ include "agent.accessKeySecret" . }}
+          {{- end }}
```
```suggestion
          {{- end }}
      {{- if eq (include "sysdig.custom_ca.useValues"  (dict "global" .Values.global.ssl "component" .Values.ssl)) "true" }}
      - name: ca-cert
        secret:
          secretName: {{ include "agent.fullname" . }}-ca
      {{- else if eq (include "sysdig.custom_ca.useExistingSecret"  (dict "global" .Values.global.ssl "component" .Values.ssl)) "true" }}
      - name: ca-cert
        secret:
          secretName: {{ include "sysdig.custom_ca.existingSecret" (dict "global" .Values.global.ssl "component" .Values.ssl) }}
      {{- else if eq (include "sysdig.custom_ca.useExistingConfigMap" (dict "global" .Values.global.ssl "component" .Values.ssl)) "true" }}
      - name: ca-cert
        configMap:
          name: {{ include "sysdig.custom_ca.existingConfigMap" (dict "global" .Values.global.ssl "component" .Values.ssl) }}
      {{- end }}
```

AlbertoBarba commented on charts/agent/templates/daemonset-windows.yaml:
```diff
@@ -0,0 +1,74 @@
+          - name: config
+            mountPath: /Sysdig/Agent/Config
+          - name: secrets
+            mountPath: /Sysdig/Agent/Secrets
```
```suggestion
            mountPath: /Sysdig/Agent/Secrets
          {{- if eq (include "sysdig.custom_ca.enabled"  (dict "global" .Values.global.ssl "component" .Values.ssl)) "true" }}
          - name: ca-cert
            mountPath: /etc/ca-certs
            readOnly: true
          {{- end }}
```

iurly committed:
add ca-certs to windows deamonset

iurly committed:
Merge remote-tracking branch 'origin/main' into SMAGENT-6917-add-support-for-windows-agent

iurly committed:
wip: readiness probe

iurly committed:
cant get probe to work

iurly committed:
readiness probe now working

iurly committed:
rework paths in the chart

- rework the configmap so to use "certificates"
  (relative) instead of /etc/ca-certs (absolute)
- adapt mountpaths accordingly
- on Windows, use relative paths, mainly for clarity

iurly committed:
wip: mount the proxy credentials secret

NOTE: this is still not implemented on the agent's side

iurly committed:
[SMAGENT-7271] Use dedicated path for AccessKey

iurly committed:
use correct secrets path for proxy credentials

iurly committed:
[SMAGENT-7299] Add livenessProbe

iurly committed:
use new healthprobe bat script

iurly committed:
fix livenessProbe timing and add it for readinessProbe

iurly committed:
Merge remote-tracking branch 'origin/main' into SMAGENT-6917-add-support-for-windows-agent

AlbertoBarba committed:
Merge branch 'main' into SMAGENT-6917-add-support-for-windows-agent

AlbertoBarba committed:
chore(agent): add newline at the end of the README file

iurly committed:
use httpGet probes instead of exec

There is a well-known memory leak in containerd
https://github.com/containerd/containerd/issues/7375
that will eventually cause the agent pod to be
restarted due to memory exhaustion.

To circumvent that, resort back to httpGet probe.
Notice how only security manager is probed
(being the most critical process) so the
connection manager remains unwatched.

iurly committed:
Merge remote-tracking branch 'origin/main' into SMAGENT-6917-add-support-for-windows-agent

iurly committed:
Merge remote-tracking branch 'origin/main' into SMAGENT-6917-add-support-for-windows-agent

iurly committed:
missed updates

iurly committed:
pin agent-windows to 1.0.0
